cmake/sysbuild/lwm2m_carrier_divided_dfu.cmake

```diff
@@ -0,0 +1,34 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+find_package(Python3 REQUIRED)
+
+function(lwm2m_carrier_divided_dfu)
+  sysbuild_get(${DEFAULT_IMAGE}_image_dir IMAGE ${DEFAULT_IMAGE} VAR APPLICATION_BINARY_DIR CACHE)
```

Comment:
does not do anything for other images e.g. mcuboot s0/s1 update?

Reply:
No. This feature should only really be used for images that exceed 256 KB and as far as I know we do not have carrier requirements concerning bootloader upgrades. Our test scope also does not cover those.

Reply:
So how does this work for applications when mcuboot is configured for directXIP mode?

Reply:
Also how does this work for other devices that might be attached to the nrf91 e.g. nrf52840 or nrf5340, considering we support updating those from the nrf91 in ncs?

Reply:
Won't work for directXIP, which is fine for our use-case. I've added a dependency on the supported swap-based variants. Also added a dependency on MCUBOOT_UPDATEABLE_IMAGES == 1.